components/experimental/src/units/measureunit.rs

```diff
@@ -58,36 +58,49 @@ impl<'data> MeasureUnitParser<'data> {
         }
 
         // If the power is found, this means that the part must start with the `-` sign.
-        match part_without_power.strip_prefix('-') {
+        match part_without_power.strip_prefix(b"-") {
             Some(part_without_power) => Ok((power, part_without_power)),
             None => Err(ConversionError::InvalidUnit),
         }
     }
 
-    fn get_si_prefix<'a>(&'a self, part: &'a str) -> (SiPrefix, &str) {
+    /// Get the SI prefix.
+    /// NOTE:
+    ///    if the prefix is not found, the function will return (SiPrefix { power: 0, base: Base::Decimal }, part).
+    fn get_si_prefix<'a>(&'a self, part: &'a [u8]) -> (SiPrefix, &[u8]) {
         let (si_prefix, part_without_si_prefix) = get_si_prefix(part);
         if part_without_si_prefix.len() == part.len() {
             return (si_prefix, part);
         }
 
-        match part_without_si_prefix.strip_prefix('-') {
+        match part_without_si_prefix.strip_prefix(b"-") {
             Some(part_without_dash) => (si_prefix, part_without_dash),
             None => (si_prefix, part_without_si_prefix),
         }
     }
 
-    /// Process a part of an identifier.
-    /// For example, if the whole identifier is: "square-kilometer-per-second",
-    /// this function will be called for "square-kilometer" with sign (1) and "second" with sign (-1).
-    fn analyze_identifier_part(
+    // TODO: add test cases for this function.
+    /// Parses a CLDR unit identifier and returns a MeasureUnit.
+    /// Examples include: `meter`, `foot`, `meter-per-second`, `meter-per-square-second`, `meter-per-square-second-per-second`, etc.
+    /// Returns:
+    ///    - Ok(MeasureUnit) if the identifier is valid.
+    ///    - Err(ConversionError::InvalidUnit) if the identifier is invalid.
+    pub fn try_from_identifier(
         &self,
-        identifier_part: &str,
-        sign: i8,
-        result: &mut Vec<MeasureUnitItem>,
-    ) -> Result<(), ConversionError> {
-        let mut identifier_part = identifier_part;
-        while !identifier_part.is_empty() {
-            let (power, identifier_part_without_power) = self.get_power(identifier_part)?;
+        identifier: &'data [u8],
+    ) -> Result<MeasureUnit, ConversionError> {
+        if identifier.starts_with(b"-") || identifier.ends_with(b"-") {
+            return Err(ConversionError::InvalidUnit);
+        }
+
+        let mut measure_unit_items = Vec::<MeasureUnitItem>::new();
+        let mut identifier = identifier;
+        let mut sign = 1;
+        while !identifier.is_empty() {
+            // First: extract the power.
+            let (power, identifier_part_without_power) = self.get_power(identifier)?;
+
+            // Second: extract the si_prefix and the unit_id.
             let (si_prefix, unit_id, identifier_part_without_unit_id) =
                 match self.get_unit_id(identifier_part_without_power) {
```

Comment:
question: why do you try parsing a unit id before the SI prefix, if you try parsing the unit ID after the SI prefix anyway?